session/session.go

```diff
@@ -11,447 +11,199 @@
 //    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 //    See the License for the specific language governing permissions and
 //    limitations under the License.
+
 package session
```

Comment:
Moved shell session to `shell.go` and recreated `session.go` with the new `Session` definition and handler interface, hence the enormous diff.